woocommerce/woocommerce-ios
# Variations: single variation deletion - UI

Part of https://github.com/woocommerce/woocommerce-ios/issues/3193.
Based on https://github.com/woocommerce/woocommerce-ios/pull/3728.

## Description

This PR adds UI integration for delete action added in https://github.com/woocommerce/woocommerce-ios/pull/3728.

## Changes

1. Enables delete action by updating `canDeleteProduct` in `ProductVariationFormViewModel`
2. Adds `ProductVariationAction.deleteProductVariation` call in `ProductVariationFormViewModel` (+ tests)
3. Moves confirmation alerts and progress views into helper extension on `ProductFormViewController` because variations deletion need different text - products are moved to trash, but variations are deleted ([reference in API doc](https://woocommerce.github.io/woocommerce-rest-api-docs/#delete-a-product-variation)).

## Test

1. Open details for product with existing variations
2. Open variations list
3. Tap on variation
4. Tap "•••" in navbar, pick "Delete"
5. Observe that variation in successfully removed

### Known limitations

Empty state is missing after removal of all variations, also parent product details is not updated. Will be fixed in separate PR.

## Video

https://user-images.githubusercontent.com/3132438/108848960-b12d3f00-75f2-11eb-9f5b-dd37c6e82c45.mp4

Update release notes:
